docs/package_templates/autotools_package/all/conanfile.py

```diff
@@ -118,17 +97,18 @@ def build_requirements(self):
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        # apply patches listed in conandata.yml
+        # Using patches is always the last resource to fix issues. If possible, try to fix the issue in the upstream project.
+        apply_conandata_patches(self)
 
     def generate(self):
-        # inject tool_requires env vars in build scope (not needed if there is no tool_requires)
-        env = VirtualBuildEnv(self)
-        env.generate()
-        # inject requires env vars in build scope
+        # inject tool_requires env vars into the build scope (not needed if there are no tool_requires)
+        VirtualBuildEnv(self).generate()
+        # inject required env vars into the build scope
         # it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable
         if not cross_building(self):
-            env = VirtualRunEnv(self)
-            env.generate(scope="build")
-        # --fpic is automatically managed when 'fPIC'option is declared
+            VirtualRunEnv(self).generate(scope="build")
+        # --fpic is automatically managed when 'fPIC' option is declared
         # --enable/disable-shared is automatically managed when 'shared' option is declared
         tc = AutotoolsToolchain(self)
```

Comment:
Can be replaced with GnuToolchain new recipes?

Reply:
Requires at least v2.3.0, though.

Would be great if this fairly simple bug in the generator could be fixed as well: https://github.com/conan-io/conan/issues/17171

Reply:
Unfortunately not. The GnuToolchain is still unstable and not documented in Conan docs.